qutip/qip/device/processor.py

```diff
@@ -637,19 +637,25 @@ def run_analytically(self, init_state=None, qc=None):
 
         Returns
         -------
-        evo_result: :class:`qutip.Result`
-            An instance of the class
-            :class:`qutip.Result` will be returned.
+        U_list: list
+            A list of propagators obtained for the physical implementation.
         """
         if init_state is not None:
             U_list = [init_state]
         else:
             U_list = []
         tlist = self.get_full_tlist()
-        # TODO replace this by get_complete_coeff
         coeffs = self.get_full_coeffs()
+
+        # Compute drift Hamiltonians
+        H = 0
+        for drift_ham in self.drift.drift_hamiltonians:
+            H += drift_ham.get_qobj(self.dims)
+
+        # Compute control Hamiltonians
         for n in range(len(tlist)-1):
-            H = sum([coeffs[m, n] * self.ctrls[m]
+            H += sum(
+                [coeffs[m, n] * self.ctrls[m]
```

Comment:
This code seems odd to me. Previously we added a separate `U` based on a separate `H` for each time `t`, but here we add to `H` each time. Do we perhaps need to create an `H_drift` above and then do `H = H_drift + sum([...])` in each loop iteration here?

Reply:
Oh, yes you are right.